### PR TITLE
TASK: Refactor and fix date converter

### PR DESCRIPTION
**Pull request by daniellienert**
The actual extracted date of the sample image (at least on my and travis's setup) lacks the time zone deviation, which makes the dateTime Object creation fail.

### CODE REVIEW

**Comment by daniellienert** [comment]
@ComiR: This time fixed for real (hopefully). Review appreciated.

File: Classes/Domain/Extractor/IptcIimExtractor.php [hunk 1]
```diff
@@ -111,21 +107,16 @@ public function extractMetaData(FlowResource $resource, MetaDataCollection $meta
...
-        if (!empty($digitalCreationDateString)) {
-            $digitalCreationTimeString = $iim->getProperty(Iptc\Iim::DIGITAL_CREATION_TIME);
-            $digitalCreationDateString .= empty($digitalCreationTimeString) ? '000000+0000' : $digitalCreationTimeString;
-            $iptcData['DigitalCreationDate'] = \DateTime::createFromFormat('YmdHisO', $digitalCreationDateString);
-        }
+        $iptcData['DigitalCreationDate'] = DateConverter::convertIso8601DateAndTimeString($iim->getProperty(Iptc\Iim::DIGITAL_CREATION_DATE), $iim->getProperty(Iptc\Iim::DIGITAL_CREATION_TIME));
         $metaDataCollection->set('iptc', new Dto\Iptc($iptcData));
     }
-
+
```
Comment (ComiR) [inline Classes/Domain/Extractor/IptcIimExtractor.php:113]: style issue

File: Classes/Domain/Extractor/IptcIimExtractor.php [hunk 2]
```diff
@@ -111,21 +107,16 @@ public function extractMetaData(FlowResource $resource, MetaDataCollection $meta
...
     }
-
+    
     /**
      * @param FlowResource $resource
      * @return array
      * @throws ExtractorException
      */
-    public function readRawIptcData(FlowResource $resource)
+    protected function readRawIptcData(FlowResource $resource)
```
Comment (ComiR) [inline Classes/Domain/Extractor/IptcIimExtractor.php:119]: could even be declared `static`.

File: Tests/Unit/Domain/Extractor/Converter/DateConverterTest.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * @test
+     * @dataProvider gpsTimeStampDataProvider
+     *
+     * @param array $timeStamp
+     * @param string $dateStamp
+     * @param \DateTime $expected
+     */
+    public function convertGpsDateAndTime($timeStamp, $dateStamp, $expected)
```
Comment (ComiR) [inline Tests/Unit/Domain/Extractor/Converter/DateConverterTest.php]: I'd switch `$timeStamp` and `$dateStamp` to match `DateConverter::convertGpsDateAndTime`.

**Review by ComiR — CHANGES_REQUESTED** [review]
Only some remarks in regard to code styling. Apart from that LGTM by reading.

**Pull request merged**